### PR TITLE
feat(functions): add Pub/Sub test examples

### PR DESCRIPTION
**Pull request by ace-n**

### CODE REVIEW

File: functions/helloworld_pubsub/SampleIntegrationTest.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    private static $process;
+
+    /** @var Client */
+    private static $client;
+
+    public function dataProvider()
+    {
+        return [
+            [
+                'cloudevent' => [
```
Comment (bshaffer) [inline functions/helloworld_pubsub/SampleIntegrationTest.php:47]: Would it be better to use the `CloudEvent1 class here?

```php
'cloudevent' => CloudEvent::fromArray([
    'id' => uniqid(),
    'source' => 'pubsub.googleapis.com',
    'specversion' => '1.0',
    'type' => 'google.cloud.pubsub.topic.v1.messagePublished',
],
```

File: functions/helloworld_pubsub/SampleIntegrationTest.php [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        // Prepare the HTTP headers for a CloudEvent.
+        $cloudEventHeaders = [];
+        foreach ($cloudevent as $key => $value) {
+            $cloudEventHeaders['ce-' . $key] = $value;
+        }
+
+        // Send an HTTP request using CloudEvent metadata.
+        $resp = self::$client->post('/', [
+            'body' => json_encode($data),
+            'headers' => $cloudEventHeaders + [
```
Comment (bshaffer) [inline functions/helloworld_pubsub/SampleIntegrationTest.php:80]: I'd rather explicitly set these, so they're more clear and readable:
```php
'headers' => [
    'ce-id' => $cloudevent->getId(),
    'ce-source' => $cloudevent->getSource(),
    'ce-specversion' => $cloudevent->getSpecVersion(),
    'ce-type' => $cloudevent->getType(),
    'content-type' => 'application/json'
],
```

**Review by bshaffer — COMMENTED** [review]
This looks great! Some readability suggestions is all I have here.

File: functions/helloworld_pubsub/SampleIntegrationTest.php (hunk 2 above)
Comment (ace-n) [reply functions/helloworld_pubsub/SampleIntegrationTest.php:80]: Done.

File: functions/helloworld_pubsub/SampleIntegrationTest.php (hunk 1 above)
Comment (ace-n) [reply functions/helloworld_pubsub/SampleIntegrationTest.php:47]: Done.

**Comment by ace-n** [comment]
@bshaffer Comments addressed. 🙂

**Review by bshaffer — APPROVED** [review]

File: functions/helloworld_pubsub/SampleIntegrationTest.php [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+ * limitations under the License.
+ */
+
+declare(strict_types=1);
+
+// [START functions_pubsub_integration_test]
+
+namespace Google\Cloud\Samples\Functions\HelloworldPubsub;
+
+use Google\CloudFunctions\CloudEvent;
```
Comment (grayside) [inline functions/helloworld_pubsub/SampleIntegrationTest.php:24]: follow-up: the docs should point out the test dependencies, perhaps showing a composer.json snippet.

One special note, is this is taking a testing dependency on the functions framework itself, we're recommending developers make that a runtime dependency anyway, but they may not expect that.

File: functions/helloworld_pubsub/SampleIntegrationTest.php [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+     * @dataProvider dataProvider
+     */
+    public function testHelloPubsub(
+        CloudEvent $cloudevent,
+        array $data,
+        string $statusCode,
+        string $expected
+    ): void {
+        // Send an HTTP request using CloudEvent metadata.
+        $resp = self::$client->post('/', [
```
Comment (grayside) [inline functions/helloworld_pubsub/SampleIntegrationTest.php:72]: nit: add comment explaining this is initialized in the startFunctionFramework method below.

File: functions/helloworld_pubsub/SampleUnitTest.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                'cloudevent' => CloudEvent::fromArray([
+                    'id' => uniqid(),
+                    'source' => 'pubsub.googleapis.com',
+                    'specversion' => '1.0',
+                    'type' => 'google.cloud.pubsub.topic.v1.messagePublished',
+                    'data' => [
+                        'data' => base64_encode('John')
+                    ],
+                ]),
+                'statusCode' => '200',
```
Comment (grayside) [inline functions/helloworld_pubsub/SampleUnitTest.php:57]: statusCode not under test. Should probably remove?

**Review by grayside — APPROVED** [review]
Approval subject to comment resolution.

File: functions/helloworld_pubsub/SampleIntegrationTest.php (hunk 4 above)
Comment (ace-n) [reply functions/helloworld_pubsub/SampleIntegrationTest.php]: Reorganized snippets as we discussed during our meeting today. (i.e. moved `startFunctionFramework` above this method.)

File: functions/helloworld_pubsub/SampleUnitTest.php (hunk 5 above)
Comment (ace-n) [reply functions/helloworld_pubsub/SampleUnitTest.php:57]: Good catch, done.

**Pull request merged**